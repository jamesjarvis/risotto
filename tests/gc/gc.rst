type Pet struct {
}

type Person struct {
    Name string;
    Pets []Pet;
}

new (p Person) () Person {
    p.Name = "TEST"
    p.Pets = []Pet{}

    return p
}

{
    println("#####################################")

    for i := 0; i < 5; i += 1 {
        ui := new Person()
    }

    vm_stats()
    gc()
    vm_stats()
}

{
    println("#####################################")

    u0 := new Person()

    {
        u1 := new Person()
        u2 := new Person()
        u3 := new Person()
    }

    vm_stats()
    gc()
    vm_stats()
}

{
    println("#####################################")

    vm_stats()

    for i := 0; i < 950; i += 1 {
        ui := new Person()
    }

    vm_stats()
    gc()
    vm_stats()

    println("===============================================")

    gc()
    vm_stats()
}
