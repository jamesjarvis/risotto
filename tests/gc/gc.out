#####################################
Objects count: 5
Max Objects: 1000
Objects count: 0
Max Objects: 1000
#####################################
Objects count: 4
Max Objects: 1000
Objects count: 1
Max Objects: 1000
#####################################
Objects count: 1
Max Objects: 1000
Objects count: 951
Max Objects: 1000
Objects count: 0
Max Objects: 1000
===============================================
Objects count: 0
Max Objects: 1000
